src/EventBus.js

```diff
@@ -1,100 +1,413 @@
+         *  EventBus.off("click",onClick,buttonScope);
+         *  EventBus.off("click",buttonScope);
+         *  EventBus.off("click").off("mousemove");
+         * @param type
+         * @param callback
+         * @param scope
+         * @returns {EventBusClass}
+         */
+        off: function (type, callback, scope) {
+            //support EventBus.off(['click',/click/]);
+            type = processMultiTypes.call(this, type, this.off, [undefined, callback, scope]);
+            if (type == this)return this;
+
+            var isRegExpType = type instanceof RegExp;
+            var eventType = isRegExpType ? type.toString() : type;
+            var newArray = [];
+            if (typeof callback == "object") {
+                scope = callback;
+                callback = undefined;
+            }
+            var allCallback = callback == undefined;
+            var allScope = scope == undefined;
+            if(EventBusClass.DEFAULT.SHOW_LOG)
+                console.log("off=>off event type:", type, " all callback:", allCallback, " all scope:", allScope);
+            // console.log("callback:", callback, " scope:", scope);
+            function isRemove(listener) {
+                return (allCallback ? true : listener.callback == callback) &&
+                    (allScope ? true : listener.scope == scope)
+            }
+
+            if (!isRegExpType) {
+                if (typeof this.listeners[eventType] != "undefined") {
+                    if (!(allCallback && allScope))
+                        iterator(this.listeners[eventType], function (i, listener) {
+                            if (!isRemove(listener)) newArray.push(listener);
+                            else {
+                                // console.log("remove event listener:", type, listener);
+                            }
+                        });
+                    this.listeners[eventType] = newArray;
+                }
+            } else {
+                iterator(this.regexListeners, function (index, listener) {
+                    if (!(listener.eventType == eventType && isRemove(listener))) newArray.push(listener);
+                    else {
+                        // console.log("remove listener:", type, listener);
+                    }
+                });
+                this.regexListeners = newArray;
+            }
+            return this;
+        },
+        /**
+         * redirect("click","onClick",function(event){return true;});
+         * redirect("click","onClick");
+         * redirect("click",function(event){return "onClick"},function(event){return true;});
+         * redirect(/\w*_click/,"onClick",/btn[0-9]+_click/);
+         * @param origin
+         * @param endpoint
+         * @param condition
+         * @param processor processor be called before event redirection
+         * @return {EventBusClass}
+         */
+        redirect: function (origin, endpoint, condition, processor) {
```

Comment:
The purpose of implementing the redirect method is to increase the renaming of the event so that the specified event is turned into other events, or a class event turns into a single event. 

Specific use cases have been included in the example/node/cs-test.js.